kratos/processes/parallel_distance_calculation_process.cpp

```diff
@@ -383,6 +390,60 @@ void ParallelDistanceCalculationProcess<TDim>::CalculateExactDistancesOnDividedE
     KRATOS_CATCH("")
 }
 
+template<unsigned int TDim>
+void ParallelDistanceCalculationProcess<TDim>::AbsDistancesOnDividedElements()
+{
+    KRATOS_TRY
+
+    //identify the list of elements divided by the original distance distribution
+    //preserving the original position of the free surface
+    array_1d<double,TDim+1> dist;
+    block_for_each(mrModelPart.Elements(), dist, [&](Element& rElement, array_1d<double,TDim+1>& rDist){
+        auto& element_geometry = rElement.GetGeometry();
+        // Set distances vector from non-historical database
+        for (unsigned int j = 0; j < TDim + 1; j++) {
+            rDist[j] = mAuxDistanceGetter(element_geometry[j]);
```

Comment:
You are duplicating a lot of code just to change this line, maybe you could add this to the if condition in https://github.com/KratosMultiphysics/Kratos/blob/dd678b33e04884aef84bd7384e86122496744f4c/kratos/processes/parallel_distance_calculation_process.cpp#L343

Reply:
It is basically not my code. I recovered the old code just to bring back the old functionality.

Reply:
It does not matter, code is not "owned" by anyone. Duplication code should be avoided in any case

Reply:
Completely agree with @ddiezrod observation.